ILAB Phase 3 Github Markdown Documentation (#254)
opendatahub-io/ilab-on-ocp: run-pipeline-on-rhoai-.md

    - `registry.redhat.io/rhelai1/prometheus-8x7b-v2-0@sha256:9fcb49c230f6e73ff944377307bb83a05ae3ac20300af75e429151f4f8bf4285`
    - `quay.io/modh/odh-generic-data-science-notebook@sha256:7c1a4ca213b71d342a2d1366171304e469da06d5f15710fab5dd3ce013aa1b73`
    - `quay.io/modh/vllm@sha256:3c56d4c2a5a9565e8b07ba17a6624290c4fb39ac9097b99b946326c09a8b40c8`
    - `quay.io/modh/vllm@sha256:97b91f9bd71202f5de8d379cfb61baec887b47f836a2ff8b158c946196de5660`
    - `quay.io/opendatahub/workbench-images@sha256:7f26f5f2bec4184af15acd95f29b3450526c5c28c386b6cb694fbe82d71d0b41`
    - `ghcr.io/oras-project/oras:main@sha256:8859e7e3ae510fb921ebeb109ac9d3e3bb91799e0d52001ae456df33929029db`

  * **500GB PersistentVolumeClaim (PVC) for Mixtral**:

    The proposed method to deploy Mixtral requires a **500GB PVC**.
    - In a **disconnected cluster**, ensure that your OpenShift environment has sufficient storage capacity and a **StorageClass** configured to provision this PVC.
    - If automatic PVC creation fails, you may need to manually create a PersistentVolume (PV) and bind it to a PVC.

>   * **Additional Git Repository for Mirroring**:
> 
>     - The iLab pipeline uses a taxonomy repo from GitHub.
>     - The following repository should be mirrored for use in a disconnected cluster:
> 
>       ```bash
>       Original repo: github.com/sallyom/taxonomy.git
>       repo_branch: utah-hockey
>       ```

Reviewer: Sorry, this was just a note in our internal document. Sally's repo was just an example we used
```suggestion
  * **Accessible git repository with the taxonomy**:

    - The iLab pipeline uses a taxonomy git repository, which should be accessible from the disconnected cluster
```

```

> [!NOTE]
> Note: If using a custom CA certificate you must provide the relevant data in a ConfigMap. The config map name and
> key are then provided as a parameter to the pipeline as well as in the `teacher-model-details-k8s-secret` secret above.

If you deployed the Teacher server model using the optional instructions above then you can retrieve `api_key` by
running the following command:

```bash
SDG_API_KEY=$(oc -n <data-science-project-name/namespace> create token mixtral-sa)
```

### (Optional) - Setup NFS StorageClass

> [!CAUTION]
> The image provided here is for test purposes only.
> Users must provide a production ready storageclass with ReadWriteMany capability.

This step is needed when the cluster doesn't have a storage provisioner capable of provisioning PersistentVolumeClaim with ReadWriteMany capability.

Installing the NFS CSI driver
```bash
$ curl -skSL https://raw.githubusercontent.com/kubernetes-csi/csi-driver-nfs/v4.9.0/deploy/install-driver.sh | bash -s v4.9.0 --`
```

For deploying an in-cluster NFS server

```bash
oc new-project nfs

> oc apply -f ./nfs-server-deployment.yaml

Reviewer: The document doesn't explain where is this file downloaded or what should be the content

Reviewer: I see now that the file is in the standalone folder. Consider adding a link in the markdown doc

```

> [!NOTE]
> Note:  Check the root PersistentVolumeclaim that'll be created and the requested storage.

For creating NFS storage-class
```bash

> oc apply -f ./nfs-storage-class.yaml

Reviewer: The document doesn't explain where is this file downloaded or what should be the content

Reviewer: I see now that the file is in the standalone folder. Consider adding a link in the markdown doc
